lib/lib.misc.php

```diff
@@ -1319,26 +1344,41 @@ function eventlog($type, $dataid, $action, $cid = null, $json = null, $id = null
 	}
 
 	// Look up external/API ID from various sources.
-	if ( $id===null ) $id = rest_extid($type, $dataid);
+	if ( $ids===null ) {
+		$ids = array_map(function($dataid) use ($type) {
+			return rest_extid($type, $dataid);
```

Comment:
Please add a TODO to also add multiple id functionality to rest_extid. Otherwise this does n DB queries which could be a single one.